src/rules/keywords-in-logical-order.ts

```diff
@@ -31,28 +38,44 @@ export function run({feature}: GherkinData): RuleError[] {
 
       if (keywordPosition < maxKeywordPosition) {
         const maxKeyword = keywordList[maxKeywordPosition];
-        // console.log(createError(step, maxKeyword));
-        errors.push(createError(step, maxKeyword));
+        errors.push(createError(step, maxKeyword, null));
       }
-
+      existsKeyword[keywordList[keywordPosition]] = true;
       maxKeywordPosition =
         Math.max(maxKeywordPosition, keywordPosition) || keywordPosition;
     });
+    if (detectMissingKeywords && child.scenario && !Object.values(existsKeyword).every((value) => value === true)) {
```

Comment:
Redudante el evaluar un boolean así
```suggestion
    if (detectMissingKeywords && child.scenario && !Object.values(existsKeyword).every((value) => value)) {
```